homeassistant/components/aemet/config_flow.py

```diff
@@ -0,0 +1,104 @@
+"""Config flow for AEMET OpenData."""
+from aemet_opendata import AEMET
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_MODE,
+    CONF_NAME,
+)
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+
+from .const import DEFAULT_FORECAST_MODE, DEFAULT_NAME, FORECAST_MODES
+from .const import DOMAIN  # pylint:disable=unused-import
+
+
+class AemetConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for AEMET OpenData."""
+
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return AemetOptionsFlow(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+            latitude = user_input[CONF_LATITUDE]
+            longitude = user_input[CONF_LONGITUDE]
+
+            await self.async_set_unique_id(f"{latitude}-{longitude}")
+            self._abort_if_unique_id_configured()
+
+            api_online = await _is_aemet_api_online(self.hass, user_input[CONF_API_KEY])
+            if not api_online:
+                errors["base"] = "invalid_api_key"
+
+            if not errors:
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME], data=user_input
+                )
+
+        schema = vol.Schema(
+            {
+                vol.Required(CONF_API_KEY): str,
+                vol.Optional(CONF_NAME, default=DEFAULT_NAME): str,
+                vol.Optional(
+                    CONF_LATITUDE, default=self.hass.config.latitude
+                ): cv.latitude,
+                vol.Optional(
+                    CONF_LONGITUDE, default=self.hass.config.longitude
+                ): cv.longitude,
+                vol.Optional(CONF_MODE, default=DEFAULT_FORECAST_MODE): vol.In(
```

Comment:
Mode should only be part of the options flow. Optional items not related specifically with the connection go there.

Reply:
But preferably we shouldn't have an option at all. See Matthew's review below.

Reply:
As above, this is inherited from OpenWeatherMap: https://github.com/home-assistant/core/blob/fd177441b3cfddb22b893b26e91862307bbcc437/homeassistant/components/openweathermap/config_flow.py#L78

Reply:
If there's no issue, please remove the option and let the user enable/disable the entities instead. Start with just the default mode entity enabled by default.

Reply:
I'll mark this as resolved and deal with it in @MatthewFlamm's review.